packages/origin-247-transfer/src/EnergyTransferRequest.ts

```diff
@@ -164,9 +233,29 @@ export class EnergyTransferRequest {
             createdAt: new Date(),
             updatedAt: new Date(),
             certificateId: null,
-            isCertificatePersisted: false,
             validationStatusRecord: {},
-            computedValidationStatus: TransferValidationStatus.Pending
+            state: BigNumber.from(params.volume).eq(0) ? State.Skipped : State.IssuanceAwaiting,
+            processError: null
         };
     }
+
+    /**
+     * This method works like super simple state machine.
+     * It ensures correct current state at runtime (because it's impossible to check this in compilation time),
+     * and ensures correct next state in compilation time (because it's much better DX).
+     *
+     * Any runtime error here indicates a bug.
+     */
+    private nextState<T extends keyof StateTransition, P extends StateTransition[T]>(
+        expectedCurrentState: T,
+        nextState: P extends (infer R)[] ? R : never
+    ) {
+        if (this.attrs.state !== expectedCurrentState) {
+            throw new Error(
+                `ETR ${this.attrs.id} is in invalid state to execute method. Expected ${expectedCurrentState}, current: ${this.attrs.state}`
```

Comment:
This effectively was throwing all the time for me during writing tests, so it serves it's purpose